test/testMdastToVDOM.js

```diff
@@ -65,10 +65,10 @@ describe('Test MDAST to VDOM Transformation', () => {
   it('Programmatic Matcher Function', () => {
     const mdast = fs.readJSONSync(path.resolve(__dirname, 'fixtures', 'simple.json'));
     const transformer = new VDOM(mdast, action.secrets);
-    transformer.match(({ type }) => type === 'heading', () => '<h1>All Headings are the same to me</h1>');
+    transformer.match(({ type }) => type === 'heading', () => '<h1 id="h1">All Headings are the same to me</h1>');
     assertTransformerYieldsDocument(
       transformer,
-      '<h1>All Headings are the same to me</h1>',
+      '<h1 id="user-content-h1">All Headings are the same to me</h1>',
```

Comment:
is this relevant to the XSS changes?

Reply:
Seems like a leak from your other branch.

Reply:
It actually is related to XSS. This is to avoid DOM clobbering.
Both `id` and `name` are exposed on the global window object, and if you have a non-secure browser (webkit is, but some other aren't), then you can actually override global properties on the window with this.

```
<h1 id="addEventListener"></h1>
<a name="removeEventListener"></h1>
```

The best practice, and also what is used by GitHub, is to always prefix these 2 properties with "user-content-" when they are coming from user input (i.e. the markdown in our case).